File: Proyecto Riojas/Proyecto Final1/Proyecto Final1/Lista.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proyecto_Final1
{
    class Lista
    {
        //Variable(s)
        private Nodo1 head;

        //Propiedades
        public Nodo1 Head
        {
            get
            {
                return head;
            }
            set
            {
                head = value;
            }
        }

        //Constructores
        public Lista()
        {
            head = null;
        }

        //Metodos
        public void Agregar(Nodo1 nuevo)
        {
            if (head == null)
            {
                head = nuevo;
                return;
            }

            Nodo1 copia = head;
            while (copia.Siguiente != null)
            {
                copia = copia.siguiente;
            }
            copia.Siguiente = nuevo;
        }
        public bool Buscar(int valor)
        {
            if (head == null)
            {
                return false;
            }

            Nodo1 copia = head;
            while (copia != null)
            {
                if (copia.Valor == valor)
                {
                    return true;
                }
                copia = copia.siguiente;
            }

            return false;
        }

        public GrafosConListas SeleccionarCamino(int valor)
        {
            if (head == null)
            {
                return null;
            }

            Nodo1 copia = head;
            while (copia != null)
            {
                if (copia.Valor == valor)
                {
                    return copia.Lazos;
                }
                copia = copia.siguiente;
            }
            return null;
        }



        public void EliminarLazos(int valor)
        {
            Nodo1 copia = head;
            while (copia != null)
            {
                GrafosConListas camino = copia.Lazos;
                camino.Eliminar(valor);

                copia = copia.siguiente;
            }
        }

        public void Eliminar(int valor)
        {
            if (head == null)
            {
                return;
            }
            if (head.Valor == valor)
            {
                head = head.siguiente;
                return;
            }

            Nodo1 copia = head;
            Nodo1 auxiliar = head;
            while (copia != null)
            {
                if (copia.Valor == valor)
                {
                    auxiliar.Siguiente = copia.Siguiente;
                    return;
                }
                auxiliar = copia;
                copia = copia.siguiente;
            }
        }

        public override string ToString()
        {
            string grafo = "";
            Nodo1 copia = head;
            while (copia != null)
            {
                grafo += "[" + copia.Valor + "]";

                GrafosConListas camino = copia.Lazos;
                grafo += camino.ToString();

                copia = copia.siguiente;
                grafo += Environment.NewLine;
            }
            return grafo;
        }
    }
}

File: ProyectovisualBasic/ProyectovisualBasic/Lista.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Proyecto_Final1

    Class Lista

        'Variable(s)
        Public head As Nodo1

        'Propiedades
        Public Property Head As Nodo1
            Get
                Return Me.head
            End Get
            Set
                Me.head = Value
            End Set
        End Property

        'Constructores
        Public Sub New()
            MyBase.New
            Me.head = Nothing
        End Sub

        'Metodos
        Public Sub Agregar(ByVal nuevo As Nodo1)
            If (Me.head Is Nothing) Then
                Me.head = nuevo
                Return
            End If

            Dim copia As Nodo1 = Me.head

            While (Not (copia.Siguiente) Is Nothing)
                copia = copia.siguiente

            End While

            copia.Siguiente = nuevo
        End Sub

        Public Function Buscar(ByVal valor As Integer) As Boolean
            If (Me.head Is Nothing) Then
                Return False
            End If

            Dim copia As Nodo1 = Me.head

            While (Not (copia) Is Nothing)
                If (copia.Valor = valor) Then
                    Return True
                End If

                copia = copia.siguiente

            End While

            Return False
        End Function

        Public Function SeleccionarCamino(ByVal valor As Integer) As GrafosConListas
            If (Me.head Is Nothing) Then
                Return Nothing
            End If

            Dim copia As Nodo1 = Me.head

            While (Not (copia) Is Nothing)
                If (copia.Valor = valor) Then
                    Return copia.Lazos
                End If

                copia = copia.siguiente

            End While

            Return Nothing
        End Function

        Public Sub EliminarLazos(ByVal valor As Integer)
            Dim copia As Nodo1 = Me.head

            While (Not (copia) Is Nothing)
                Dim camino As GrafosConListas = copia.Lazos
                camino.Eliminar(valor)
                copia = copia.siguiente

            End While

        End Sub

        Public Sub Eliminar(ByVal valor As Integer)
            If (Me.head Is Nothing) Then
                Return
            End If

            If (Me.head.Valor = valor) Then
                Me.head = Me.head.siguiente
                Return
            End If

            Dim copia As Nodo1 = Me.head
            Dim auxiliar As Nodo1 = Me.head

            While (Not (copia) Is Nothing)
                If (copia.Valor = valor) Then
                    auxiliar.Siguiente = copia.Siguiente
                    Return
                End If

                auxiliar = copia
                copia = copia.siguiente

            End While

        End Sub

        Public Overrides Function ToString() As String
            Dim grafo As String = ""
            Dim copia As Nodo1 = Me.head

            While (Not (copia) Is Nothing)
                grafo = (grafo + ("[" _
                            + (copia.Valor + "]")))
                Dim camino As GrafosConListas = copia.Lazos
                grafo = (grafo + camino.ToString)
                copia = copia.siguiente
                grafo = (grafo + Environment.NewLine)

            End While

            Return grafo
        End Function
    End Class
End Namespace